src/lib.rs

```diff
@@ -109,13 +117,13 @@ impl<'a> WccRunner<'a> {
         self,
         project_path: &Path,
         grcov_file: GrcovFile<P>,
-        json_path: &Path,
     ) -> Result<WccOutput> {
         // Check if json_path is a json file.
-        if json_path
-            .extension()
-            .map(|ext| ext.to_ascii_lowercase())
-            .map_or(true, |ext| ext != "json")
+        if self
+            .0
+            .json_path
+            .and_then(|path| path.extension().map(|ext| ext.to_ascii_lowercase()))
+            .map_or(false, |ext| ext != "json")
```

Comment:
We can use `map_or(false, |path| path.extension().map_or(false, |ext| ext.to_ascii_lowercase() != "json"))` perhaps

Reply:
Sure!